src/main/java/com/amazon/aws/iot/greengrass/component/common/DependencyProperties.java

```diff
@@ -19,13 +19,13 @@
 public class DependencyProperties {
     @JsonSerialize(using = RequirementSerializer.class)
     @NonNull
-    Requirement versionRequirement;
+    Range versionRequirement;
```

Comment:
Are Requirements and Ranges equivalent?

Reply:
Yes, that's the semver term for what the other library called a requirement.

Reply:
I'd recommend downloading this branch and then writing test code to play around with it an check whatever you think is needed. Just a code review with this volume of code is unlikely to find anything.